src/native/libs/System.Security.Cryptography.Native/openssl.c

```diff
@@ -684,13 +684,13 @@ BIO* CryptoNative_GetX509NameInfo(X509* x509, int32_t nameType, int32_t forIssue
                     {
                         BIO* b = BIO_new(BIO_s_mem());
                         ASN1_STRING_print_ex(b, str, ASN1_STRFLGS_UTF8_CONVERT);
-                        sk_GENERAL_NAME_free(altNames);
+                        GENERAL_NAMES_free(altNames);
                         return b;
                     }
                 }
             }
 
-            sk_GENERAL_NAME_free(altNames);
+            GENERAL_NAMES_free(altNames);
```

Comment:
I was expecting sk_GENERAL_NAMES_pop_free.  What's the difference?

Reply:
`GENERAL_NAMES` is a typedef for a STACK_OF GENERAL_NAME. https://github.com/openssl/openssl/blob/4a3e8f08306c64366318e26162ae0a0eb7b1a006/include/openssl/x509v3.h.in#L189

Seeing exactly how `GENERAL_NAMES_free` is implemented is difficult since all of this is Perl-generated goo, but a quick disassembly shows that eventually they both lead up to the same ASN1_OBJECT cleanup calls. the pop_free just means you end up passing the GENERAL_NAME_free as a callback to cleanup individual items.

I ultimately settled on this because that's what OpenSSL is doing, and we using this elsewhere at 

https://github.com/dotnet/runtime/blob/82a0456a110d5db1e4e306df32bf2a8ce856d7d1/src/native/libs/System.Security.Cryptography.Native/apibridge.c#L703

I also ran my allocation test with this changes, and I did not observe a leak.